### ISSUE TITLE
Question about the purpose of torch.clamp(targets - 1, 0) in prepare_ctc_scores

### ISSUE THREAD

**Issue by TiloccaS** [marcpaga/basecalling_architectures#10]
Hi,

In prepare_ctc_scores, the line:

```

def prepare_ctc_scores(self, scores, targets):
--
194 | # convert from CTC targets (with blank=0) to zero indexed
195 | targets = torch.clamp(targets - 1, 0)


```

seems to map both the CTC blank (0) and the first base (e.g., 'A') to index 0.
How is the model expected to distinguish between them later?

Wouldn’t this create ambiguity during the k-mer state encoding?

Thanks for the clarification!

**Comment by marcpaga** [comment]
Hi,

This is done on the targets, and since they should be length padded with blanks/zeros, an example target `ACGT---` would become `ACGTAAA` as you indicate. This is not a problem, since we know the length of each sample so we can mask out the last 3 `A`s. 

I must admit as to why this is necessary I am not sure, my intuition is that you want the targets to follow the encoding starting from zero, since despite having blank as a zero in the output, they cannot be part of your target since you do not know where the model will introduce blanks.
